shell/components/cards/AppCard.vue

```diff
@@ -0,0 +1,499 @@
+<script setup>
+import LazyImage from '@shell/components/LazyImage';
+
+const emit = defineEmits(['card-click', 'repo-click', 'category-click', 'tag-click']);
+
+defineProps({
+  title: {
+    type:     String,
+    required: true
+  },
+  description: {
+    type:     String,
+    required: true
+  },
+  logo: {
+    type:    String,
+    default: ''
+  },
+  logoAltText: {
+    type:    String,
+    default: 'logo'
+  },
+  featured: {
```

Comment:
In order to make this generic we should remove hard coded references to helm app specific things like feature, deprecated, etc. The card should then accept configuration to populate those areas instead. Maybe an array for each that's a collection of 'things' like pills, links, etc. It's probably worth defining an object that caters for the known possibilities. For example

- id - unique id that we could use for testid
- label: { key - l10n key, text - text instead of key }
- link: { href, route}
- pill: { color: aaa } 

Consider the above from a ui extension developer's perspective and what structure would be easiest for them to understand / easily implement.